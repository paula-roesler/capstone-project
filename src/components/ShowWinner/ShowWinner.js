import { Link } from 'react-router-dom'
import styled from 'styled-components/macro'
import Button from '../Button'

export default function ShowWinner({
  visible,
  title,
  name,
  score,
  resetPlayers,
  players,
}) {
  const winner = players.sort((a, b) => a.score - b.score)
  return (
    <>
      <WrapperWinner hidden={visible}>
        <h1>{title}</h1>
        <div>{name}</div>
        <div>{score}</div>
      </WrapperWinner>
      <Button as={Link} to="/" onClick={resetPlayers}>
        Play again
      </Button>
    </>
  )
}

export const WrapperWinner = styled.div`
  display: grid;
  gap: 10px;
  padding: 10px;
  text-align: center;
  color: var(--primary);
  height: 100vh;
  background-color: var(--secondary);
`
