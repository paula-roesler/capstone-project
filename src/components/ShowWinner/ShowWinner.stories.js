import ShowWinner from './ShowWinner'

export default {
  title: 'ShowWinner',
  component: 'ShowWinner',
}

export const templateShowWinner = () => <ShowWinner />
