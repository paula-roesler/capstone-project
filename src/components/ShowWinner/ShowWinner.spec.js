import { screen, render } from '@testing-library/react'
import ShowWinner from '../ShowWinner'

describe('ShowWinner', () => {
  it('renders the player with the lowest score, also if there are more than one', () => {
    const { container } = render(
      <ShowWinner
        newWinners={[
          { name: 'Sue', overAllScore: 3 },
          { name: 'Joe', overAllScore: 3 },
          { name: 'Izzy', overAllScore: 4 },
        ]}
      />
    )
    expect(screen.getByText('Sue')).toBeInTheDocument()
    expect(screen.getByText('Joe')).toBeInTheDocument()
    expect(screen.queryByText('Izzy')).not.toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  screen.debug()
})
