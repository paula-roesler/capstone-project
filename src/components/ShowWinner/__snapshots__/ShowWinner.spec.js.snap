// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowWinner renders the player with the lowest score, also if there are more than one 1`] = `
<div
  class="ShowWinner__WrapperWinner-sc-1ryfj85-0 jRasyf"
>
  <h1 />
  <div
    class="ShowWinner__Winner-sc-1ryfj85-1 cTwPon"
  >
    <h3
      class="ShowWinner__WinnerName-sc-1ryfj85-2 gqLonJ"
    >
      Sue
    </h3>
    <h3
      class="ShowWinner__WinnerScore-sc-1ryfj85-3 knBFNn"
    >
      3
    </h3>
  </div>
  <div
    class="ShowWinner__Winner-sc-1ryfj85-1 cTwPon"
  >
    <h3
      class="ShowWinner__WinnerName-sc-1ryfj85-2 gqLonJ"
    >
      Joe
    </h3>
    <h3
      class="ShowWinner__WinnerScore-sc-1ryfj85-3 knBFNn"
    >
      3
    </h3>
  </div>
</div>
`;
